qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentOptionDefaults.cpp

```diff
@@ -5,56 +5,100 @@
-                const ExperimentOptionDefaults &rhs) {
-  return lhs.AnalysisMode == rhs.AnalysisMode &&
-         lhs.PolarizationAnalysis == rhs.PolarizationAnalysis &&
-         lhs.CRho == rhs.CRho && lhs.CAlpha == rhs.CAlpha &&
-         lhs.CAp == rhs.CAp && lhs.CPp == rhs.CPp &&
-         lhs.TransRunStartOverlap == rhs.TransRunStartOverlap &&
-         lhs.TransRunEndOverlap == rhs.TransRunEndOverlap &&
-         lhs.MomentumTransferMin == rhs.MomentumTransferMin &&
-         lhs.MomentumTransferMax == rhs.MomentumTransferMax &&
-         lhs.MomentumTransferStep == rhs.MomentumTransferStep &&
-         lhs.ScaleFactor == rhs.ScaleFactor &&
-         lhs.ProcessingInstructions == rhs.ProcessingInstructions &&
-         lhs.ReductionType == rhs.ReductionType &&
-         lhs.IncludePartialBins == rhs.IncludePartialBins &&
-         lhs.SummationType == rhs.SummationType &&
-         lhs.StitchParams == rhs.StitchParams;
+
+// unnamed namespace
+namespace {
+Mantid::Kernel::Logger g_log("Reflectometry GUI");
+
+std::string stringValueOrEmpty(boost::optional<double> value) {
+  return value ? std::to_string(*value) : "";
+}
+
+Experiment
+getExperimentDefaults(Mantid::Geometry::Instrument_const_sptr instrument) {
+  auto defaults = OptionDefaults(instrument);
+
+  auto analysisMode = analysisModeFromString(defaults.getStringOrDefault(
+      "AnalysisMode", "AnalysisMode", "PointDetectorAnalysis"));
+  auto reductionType = reductionTypeFromString(
+      defaults.getStringOrDefault("ReductionType", "ReductionType", "Normal"));
+  auto summationType = summationTypeFromString(defaults.getStringOrDefault(
+      "SummationType", "SummationType", "SumInLambda"));
+  auto includePartialBins =
+      defaults.getBoolOrFalse("IncludePartialBins", "IncludePartialBins");
+  auto debug = defaults.getBoolOrFalse("Debug", "Debug");
+
+  auto polarizationCorrectionType =
+      polarizationCorrectionTypeFromString(defaults.getStringOrDefault(
+          "PolarizationAnalysis", "PolarizationAnalysis", "None"));
+  auto polarizationCorrections =
+      PolarizationCorrections(polarizationCorrectionType);
+
+  auto floodCorrectionType =
+      floodCorrectionTypeFromString(defaults.getStringOrDefault(
+          "FloodCorrection", "FloodCorrection", "Workspace"));
+  auto floodWorkspace = defaults.getOptionalValue<std::string>(
+      "FloodWorkspace", "FloodWorkspace");
+  auto floodCorrections = FloodCorrections(floodCorrectionType, floodWorkspace);
+
+  auto transmissionRunRange = RangeInLambda(
+      defaults.getDoubleOrZero("StartOverlap", "TransRunStartOverlap"),
+      defaults.getDoubleOrZero("EndOverlap", "TransRunEndOverlap"));
+
+  if (!transmissionRunRange.isValid(false))
+    throw std::invalid_argument("Transmission run overlap range is invalid");
+
+  // We currently don't specify stitch parameters in the parameters file
+  // although we
+  auto stitchParameters = std::map<std::string, std::string>();
```

Comment:
This comment is unfinished, is this going to be specified in the future?

Reply:
I can't remember what I was going to write here so I've removed the unfinished bit